flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/RecordWriter.java

```diff
@@ -34,10 +34,12 @@
     void write(T record) throws Exception;
 
     /**
-     * Compact all files related to the writer. Note that compaction process is only submitted and
-     * may not be completed when the method returns.
+     * Compact files related to the writer. Note that compaction process is only submitted and may
+     * not be completed when the method returns.
+     *
+     * @param fullCompaction whether to trigger full compaction or just normal compaction
```

Comment:
I think there are two things:
- Is this compaction full or normal.
- Should we wait latest compaction.

I think we can separate these two things.

Reply:
From the point of a separated compact job, we should always wait for last compaction to complete. Otherwise new changes may never be compacted.

Reply:
You are right